File: frontend/src/service/LessonService.ts
import {EntityServiceBase} from "@/service/EntityServiceBase";
import {LessonDto} from "@/model/LessonDto";

export class LessonService extends EntityServiceBase {

    constructor() {
        super('/lesson');
    }

    public async getLessons() {
        return await super.get<LessonDto[]>('/');
    }

    public async getLesson(id: string) {
        return await super.get<LessonDto>(`/${id}`);
    }
}

File: backend/service/src/main/java/pl/pwr/ite/bd2/service/LessonService.java
package pl.pwr.ite.bd2.service;

import org.springframework.stereotype.Service;
import pl.pwr.ite.bd2.model.entity.Lesson;
import pl.pwr.ite.bd2.model.filter.LessonFilter;

public interface LessonService extends FilterableEntityService<Lesson, LessonFilter> {
}
